### PR TITLE
MGMT-8146: fix dual-stack installations

### PR DESCRIPTION
**Pull request by osherdp**
This is done by using dual-stacked networks (dah)
Also, I changed the logic setting advanced networking configurations to be more clear.

### CODE REVIEW

File: discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py [hunk 1]
```diff
@@ -211,11 +210,33 @@ def get_primary_machine_cidr(self):
...
 
     def get_provisioning_cidr(self):
-        # In dualstack/IPv6 mode the secondary network is IPv6
-        if self.is_ipv6:
+        # In dualstack mode the primary network is IPv4
+        if self.is_ipv6 and not self.is_ipv4:
             return self._config.net_asset.provisioning_cidr6
         return self._config.net_asset.provisioning_cidr
 
+    def get_all_machine_addresses(self):
```
Comment (mkowalski) [inline discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py:218]: To help ourselves from the future, can you add here a comment that this function gets **one** IPv4 and **one** IPv6 subnets? We know it now, but in some months we may be confused why out of multiple subnets the function returns only one

File: discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py [hunk 2]
```diff
@@ -211,11 +210,33 @@ def get_primary_machine_cidr(self):
...
+
+        if self.is_ipv4:
+            addresses.append(self._config.net_asset.machine_cidr)
+
+        if self.is_ipv6:
+            addresses.append(self._config.net_asset.machine_cidr6)
+
+        return addresses
+
+    def get_all_provisioning_addresses(self):
```
Comment (mkowalski) [inline discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py:229]: As above, future us will be thankful for a comment that it's not "get all" but "get one of each"

File: discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py (hunk 1 above)
Comment (osherdp) [reply discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py:218]: sure, done

File: discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py (hunk 2 above)
Comment (osherdp) [reply discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py:229]: sure, done

File: discovery-infra/test_infra/helper_classes/config/cluster_config.py [hunk 3]
```diff
@@ -21,5 +21,8 @@ class BaseClusterConfig(BaseEntityConfig, ABC):
     vip_dhcp_allocation: bool = None
     cluster_networks: List[models.ClusterNetwork] = None
     service_networks: List[models.ServiceNetwork] = None
+    machine_networks: List[models.MachineNetwork] = None
     kubeconfig_path: str = None
     network_type: str = None
+    api_vip: str = None
+    ingress_vip: str = None
```
Comment (eliorerz) [inline discovery-infra/test_infra/helper_classes/config/cluster_config.py:28]: Not sure those 2 need to be in `ClusterConfig` object

Comment (osherdp) [reply discovery-infra/test_infra/helper_classes/config/cluster_config.py:28]: I don't have a lot of choice here
Once it's in advanced networking configuration we try to save it to this config object

File: discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (osherdp) [reply discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py:230]: done

File: discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (osherdp) [reply discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py:218]: done

**Comment by osherdp** [comment]
/test e2e-metal-assisted-kube-api-late-binding-single-node
(known flaky test)

**Review by eliorerz — APPROVED** [review]
LGTM

**Pull request merged**